r/tests/testthat/test-dplyr-string-functions.R

```diff
@@ -297,6 +297,33 @@ test_that("strsplit and str_split", {
 
 })
 
+test_that("arrow_*_split_whitespace functions", {
+
+  # use only ASCII whitespace characters
+  df_ascii <- tibble(x = c("Foo\nand bar", "baz\tand qux and quux"))
+
+  # use only non-ASCII whitespace characters
+  df_utf8 <- tibble(x = c("Foo\u00A0and\u2000bar", "baz\u2006and\u1680qux\u3000and\u2008quux"))
```

Comment:
I picked these non-ASCII whitespace characters from this list: https://stackoverflow.com/a/46637343/375432
Some of the characters there were not recognized as whitespace by `arrow_utf8_split_whitespace`, in particular the zero-width ones, so I didn't use those here.